mypy/binder.py

```diff
@@ -15,7 +15,7 @@
 BindableExpression = Union[IndexExpr, MemberExpr, NameExpr]
 
 
-class Frame(Dict[Key, Type]):
+class Frame(Dict[Optional[Key], Type]):
```

Comment:
You can't just remove the `Optional` here. I'm not sure that the value actually occurs (it does not make much sense to me), but the type is used in 5 different places*,
I tried to avoid adding `if key is not None` to keep the change minimal.

(*) lines 109, 116, 151, 270, 271